lens-cube/src/main/java/org/apache/lens/cube/parse/StorageCandidate.java

```diff
@@ -518,13 +575,44 @@ private boolean evaluateMeasuresCompleteness(TimeRange timeRange) throws LensExc
   }
 
   @Override
-  public Set<FactPartition> getParticipatingPartitions() {
-    return null;
+  public boolean isExpressionEvaluable(ExpressionResolver.ExpressionContext expr) {
+    return expr.isEvaluable(this);
   }
 
   @Override
-  public boolean isExpressionEvaluable(ExpressionResolver.ExpressionContext expr) {
-    return expr.isEvaluable(this);
+  public void updateAnswerableSelectColumns(CubeQueryContext cubeql) throws LensException {
+    // update select AST with selected fields
+    int currentChild = 0;
+    for (int i = 0; i < cubeql.getSelectAST().getChildCount(); i++) {
+      ASTNode selectExpr = (ASTNode) queryAst.getSelectAST().getChild(currentChild);
```

Comment:
Should we skip measure columns that are not part of "answerableMeasureIndices" ?